src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
+type Color = 'red' | 'green' | 'blue';
+type Callback = (arg: number) => number;
 
-export class Triangle {
+const callback: Callback = (n) => Math.floor(n * 100) / 100;
 
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Ooops');
+    }
+
+    if ((a + b) <= c
+    || (a + c) <= b || (c + b) <= a) {
+      throw new Error('Ooops');
+    }
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
 
+    const area = Math.sqrt(s * ((s - this.a) * (s - this.b) * (s - this.c)));
+
+    return callback(area);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Ooops');
+    }
```

Comment:
You do this check in every class
Will be better to write separate function that can accept any number of params and check if all of them > 0